### DISCUSSION TITLE
Subpixel inference

### DISCUSSION THREAD

**Discussion by mfkeles** [talmolab/sleap#1198] [Help!] [answered]
I noticed that the predictions often have multi-pixel jumps and when overlaid these predictions form a grid (suggesting low resolution on predictions). I was wondering if there is an option that would allow subpixel inference. Or maybe this is related to how I train my network?

**Comment by roomrys** [comment, marked as answer]
Hi @mfkeles,

This we do have subpixel inference implemented in `find_global_peaks` which finds the subpixel peak of each Confidence Map (CM)...
https://github.com/talmolab/sleap/blob/5b3b598f9011826d67dbe146c7302f73cbdb0f43/sleap/nn/peak_finding.py#L338-L364
However! this function uses a fixed window size `integral_patch_size = 5` for finding the peak of each CM. That being said, if the distribution of a CM is too large (does not fit in this window), then the estimated peak value will be off. Likewise, if the window size is too large and contains multiple CM distributions (as could occur in the bottom-up pipeline), then the estimated peak value will also be off. 

To improve on this, we could expose the window size to the GUI or use a better-than-constant value to determine the window size (s.a. 3*sigma of the CM distribution). 

Until then, the parameter which user's can readily tune is the sigma of the CM's (in the Training Pipeline - Model Configuration):
![image](https://user-images.githubusercontent.com/38435167/222855208-af533337-ddca-4f92-9da7-7e50807bbf04.png)
Again, it could be that the CM distribution is too large, so a smaller sigma should help. If the smaller sigma does not seem to help, then the alternative is that the window is too large (if your model is bottom-up) which will require some code changes.

Thanks,
Liezl

**Reply by mfkeles** [reply]
Thank you @roomrys! Super helpful as always. 

Best,
Mehmet
